upgrade/preflight.php

```
@@ -141,46 +196,308 @@ function tplScannerForm($parameters=null)
+
+    // Only real template extensions may be scanned or repaired.
+    $allowedExt = ['tpl', 'html'];
+    if ('' === $template_ext) {
+        foreach ($allowedExt as $ext) {
+            $scanner->addExtension($ext);
+        }
+    } elseif (in_array(strtolower($template_ext), $allowedExt, true)) {
+        $scanner->addExtension(strtolower($template_ext));
+    } else {
+        return $output; // disallowed extension — scan nothing
+    }
```

PHPMD (2.15.0)

[error] The method smartyRunScanner uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them. (undefined)

(ElseExpression)